public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/class-feedback-list-table.php

```diff
@@ -0,0 +1,259 @@
+		remove_filter( 'comment_author', array( $this, 'floated_admin_avatar' ), 10 );
+	}
+
+	/**
+	 * Get the screen option for items per page.
+	 *
+	 * @param string $comment_status Unused.
+	 *
+	 * @return int
+	 */
+	public function get_per_page( $comment_status = 'all' ) {
+		/** @global string $page_hook */
+		global $page_hook;
+
+		$option = str_replace( '-', '_', $page_hook ) . '_per_page';
+
+		return $this->get_items_per_page( $option );
+	}
+
+	/**
+	 * Other controls above/below the list table besides bulk actions.
+	 *
+	 * @param string $which
+	 *
+	 * @return void
+	 */
+	protected function extra_tablenav( $which ) {
+		// This omits the comment type filter.
+		parent::extra_tablenav( '' );
+	}
+
+	/**
+	 * Define list table columns.
+	 *
+	 * @return array
+	 */
+	public function get_columns() {
+		global $post_id;
+
+		$columns = array();
+
+		if ( $this->checkbox ) {
+			$columns['cb'] = '<input type="checkbox" />';
+		}
+
+		$columns['name']     = _x( 'Submitted by', 'column name', 'wordcamporg' );
+		$columns['feedback'] = _x( 'Feedback', 'column name', 'wordcamporg' );
+		$columns['rating']   = _x( 'Rating', 'column name', 'wordcamporg' );
+
+		if ( ! $post_id ) {
+			/* translators: Column name or table row header. */
+			$columns['response'] = _x( 'In Response To', 'column name', 'wordcamporg' );
+		}
+
+		$columns['date'] = _x( 'Submitted On', 'column name', 'wordcamporg' );
+
+		return $columns;
+	}
+
+	/**
+	 * Which columns are sortable.
+	 *
+	 * @return array
+	 */
+	protected function get_sortable_columns() {
+		return array(
+			'name'     => 'comment_author',
+			'rating'   => array( 'rating', true ), // Second array item `true` means sort will descend first.
+			'response' => 'comment_post_ID',
+			'date'     => 'comment_date',
+		);
+	}
+
+
+	/**
+	 * Get the name of the default primary column.
+	 *
+	 * @return string Name of the default primary column.
+	 */
+	protected function get_default_primary_column_name() {
+		return 'feedback';
+	}
+
+	/**
+	 * Render the bulk edit checkbox.
+	 *
+	 * @param WP_Comment $comment
+	 *
+	 * @return void
+	 */
+	public function column_cb( $comment ) {
+		if ( current_user_can( 'moderate_' . COMMENT_TYPE ) ) {
+			?>
+			<label class="screen-reader-text" for="cb-select-<?php echo esc_attr( $comment->comment_ID ); ?>"><?php esc_html_e( 'Select feedback', 'wordcamporg' ); ?></label>
+			<input id="cb-select-<?php echo esc_attr( $comment->comment_ID ); ?>" type="checkbox" name="bulk_edit[]" value="<?php echo esc_attr( $comment->comment_ID ); ?>" />
+			<?php
+		}
+	}
+
+	/**
+	 * Render the Name column.
+	 *
+	 * @param WP_Comment $comment
+	 *
+	 * @return void
+	 */
+	public function column_name( $comment ) {
+		$feedback = get_feedback_comment( $comment );
+
+		echo '<strong>';
+		comment_author( $comment );
+		echo '</strong>';
+
+		$user = ( $feedback->user_id ) ? new WP_User( $feedback->user_id ) : null;
+
+		if ( $user ) {
```

Comment:
If the user isn't a wp.org user, should we output the email here?

Reply:
Ah good call. 3c2671ad